corehq/apps/locations/views.py

```diff
@@ -758,3 +758,36 @@ def sync_openlmis(request, domain):
     # todo: error handling, if we care.
     bootstrap_domain_task.delay(domain)
     return HttpResponse('OK')
+
+
+@locations_access_required
+def child_locations_for_select2(request, domain):
+    id = request.GET.get('id')
+    query = request.GET.get('name', '').lower()
+    user = request.couch_user
+
+    def loc_to_payload(loc):
+        return {'id': loc.location_id, 'name': loc.display_name}
+
+    if id:
+        try:
+            loc = SQLLocation.objects.get(location_id=id)
+        except SQLLocation.DoesNotExist:
+            return json_response(
+                {'message': 'no location with id %s found' % id},
+                status_code=404,
+            )
+        else:
+            return json_response(loc_to_payload(loc))
+    else:
+        locs = SQLLocation.objects.filter(domain=domain)
+        if query:
+            locs = locs.filter(name__icontains=query)
+
+        user_loc = user.get_location(domain)
+        if user_loc and not user_can_edit_any_location(user, request.project):
+            user_loc = user_loc.sql_location
+            child_locs = [loc.id for loc in user_loc.get_descendants(True)]
+            locs = locs.filter(pk__in=child_locs)
```

Comment:
This feels weird.  I'd modify the initial query if this if statement is executed.  Something like:
```python
if user_can_edit_any_location(user, request.project):
    locs = SQLLocation.objects.filter(domain=domain)
elif user_loc:
    locs = user_loc.get_descendants(include_self=True)
else:
   # I'm not sure what's appropriate here, return no locations, maybe?
```